cpp/arcticdb/processing/operation_dispatch_binary.hpp

```diff
@@ -353,24 +372,26 @@ VariantData binary_operator(const Column& col, const Value& val, Func&& func) {
                 constexpr auto output_data_type = data_type_from_raw_type<ReversedTargetType>();
                 output_column = std::make_unique<Column>(make_scalar_type(output_data_type), true);
                 Column::transform<typename col_type_info::TDT, ScalarTagType<DataTypeTag<output_data_type>>>(
-                        col,
+                        *(col.column_),
                         *output_column,
-                        [&func, raw_value](auto input_value) -> ReversedTargetType {
-                    return func.apply(raw_value, input_value);
+                        [&func, &col, &column_name, raw_value](auto input_value) -> ReversedTargetType {
+                            column_name = binary_operation_column_name(fmt::format("{}", raw_value), func, col.column_name_.value);
```

Comment:
This cannot go here, this lambda is called for every value in the column

Reply:
Fixed in [a907db9](https://github.com/man-group/ArcticDB/pull/1557/commits/a907db94c4366371434d048c630f796fcbc7a60e)